File: Estructura_de_datos_Java/heapSort.java
public class heapSort {
    static void heapify(int arr[], int n, int i) {
        int mayor = i;
        int izq = 2 * i + 1;
        int der = 2 * i + 2;

        if (izq < n && arr[izq] > arr[mayor])
            mayor = izq;

        if (der < n && arr[der] > arr[mayor])
            mayor = der;

        if (mayor != i) {
            int temp = arr[i];
            arr[i] = arr[mayor];
            arr[mayor] = temp;

            heapify(arr, n, mayor);
        }
    }

    static void heapSort(int arr[]) {
        int n = arr.length;

        for (int i = n / 2 - 1; i >= 0; i--)
            heapify(arr, n, i);

        for (int i = n - 1; i >= 0; i--) {
            int temp = arr[0];
            arr[0] = arr[i];
            arr[i] = temp;

            heapify(arr, i, 0);
        }
    }

    public static void main(String[] args) {
        int arr[] = {34, 7, 23, 32, 5, 62};

        System.out.print("Original: ");
        for (int x : arr) System.out.print(x + " ");
        System.out.println();

        heapSort(arr);

        System.out.print("Ordenado: ");
        for (int x : arr) System.out.print(x + " ");
    }
}

File: Ordenamientos_C#/heapSort.cs
using System;

class heapSort
{
    static void HeapSort(int[] arr)
    {
        int n = arr.Length;

        // Construir el heap (montículo máximo)
        for (int i = n / 2 - 1; i >= 0; i--)
            Heapify(arr, n, i);

        // Extraer uno por uno los elementos del heap
        for (int i = n - 1; i >= 0; i--)
        {
            // Mover la raíz (máximo) al final
            int temp = arr[0];
            arr[0] = arr[i];
            arr[i] = temp;

            // Llamar heapify al heap reducido
            Heapify(arr, i, 0);
        }
    }

    static void Heapify(int[] arr, int n, int i)
    {
        int mayor = i;
        int izquierda = 2 * i + 1;
        int derecha = 2 * i + 2;

        if (izquierda < n && arr[izquierda] > arr[mayor])
            mayor = izquierda;

        if (derecha < n && arr[derecha] > arr[mayor])
            mayor = derecha;

        if (mayor != i)
        {
            int temp = arr[i];
            arr[i] = arr[mayor];
            arr[mayor] = temp;

            Heapify(arr, n, mayor);
        }
    }

    static void Main()
    {
        int[] numeros = { 34, 7, 23, 32, 5, 62 };
        Console.WriteLine("Original: " + string.Join(", ", numeros));
        HeapSort(numeros);
        Console.WriteLine("Ordenado: " + string.Join(", ", numeros));
    }
}
